string name
uint64 vid
uint64 pid